[tree view] Add a method in the `apiRef` to toggle the editing status of an item (#17768)
mui/mui-x: docs/data/tree-view/rich-tree-view/editing/editing.md

```ts
apiRef.current.updateItemLabel(
  // The id of the item to update
  itemId,
  // The new label of the item
  newLabel,
);
```

{{"demo": "ApiMethodUpdateItemLabel.js"}}

### Change edition mode of an item

> Use the `setEditedItem()` API method to enter or exit editing mode for a given item.

Reviewer: ```suggestion
Use the `setEditedItem()` API method to set which item is being edited.
```

Not sure.
I think the sentence should make clear that we don't have multi editing

Author: That's a good point 🎯
